### PR TITLE
feat(oas): identify required fields in responses - store

### PR DESCRIPTION
**Pull request by patrick-medusajs**
## Scope

Store routes

## What

Add OAS `required` to response schemas.

## Why

Code generator can use the `required` property of a schema to mark fields as optional or not when generating TS types.

## Test

* Run `yarn build`
* Run `yarn openapi:generate`
* Run `yarn redocly preview-docs docs/api/store/openapi.yaml --config=docs-util/redocly/config.yaml`
* Open the documentation preview URL in a browser (http://127.0.0.1:8080)
* Expect responses to have their fields declared as `required`

### CODE REVIEW

File: packages/medusa/src/api/routes/store/carts/index.ts [hunk 1]
```diff
@@ -207,8 +212,6 @@ export type StoreCompleteCartRes =
       data: Swap
     }
 
-export type StoreCartsDeleteRes = DeleteResponse
-
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/store/carts/index.ts:211]: note: I did not find any usage throughout the codebase.

File: packages/medusa/src/api/routes/store/products/index.ts [hunk 2]
```diff
@@ -124,19 +126,25 @@ export type StoreProductsRes = {
...
+ *   - hits
  * properties:
  *   hits:
- *     type: array
  *     description: Array of results. The format of the items depends on the search engine installed on the server.
+ *     type: array
  */
 export type StorePostSearchRes = {
   hits: unknown[]
-  [k: string]: unknown
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/store/products/index.ts:134]: note: I could not find usage or example of this field in the codebase.

**Review by patrick-medusajs — COMMENTED** [review]
self-review: highlighting note worthy code changes

File: packages/medusa/src/api/routes/store/product-tags/list-product-tags.ts [hunk 3]
```diff
@@ -105,19 +108,7 @@ import { IsType } from "../../../../utils/validators/is-type"
...
- *            count:
- *              type: integer
- *              description: The total number of items available
- *            offset:
- *              type: integer
- *              description: The number of items skipped before these items
- *            limit:
- *              type: integer
- *              description: The number of items per page
+ *          $ref: "#/components/schemas/StoreProductTagsListRes"
```
Comment (olivermrbl) [inline packages/medusa/src/api/routes/store/product-tags/list-product-tags.ts:111]: praise: nice!

File: packages/medusa/src/api/routes/store/products/index.ts (hunk 2 above)
Comment (olivermrbl) [reply packages/medusa/src/api/routes/store/products/index.ts:134]: The type was added as we cannot control the response from installed search services. We require `hits`, but usually, they return additional properties specific to the provider.

As an example, the SS below is a search hit from an Algolia installation:

CleanShot 2023-02-17 at 09 46 54

**Review by olivermrbl — APPROVED** [review]
Great work as well! 💪

**Review by shahednasser — APPROVED** [review]
LGTM

File: packages/medusa/src/api/routes/store/products/index.ts (hunk 2 above)
Comment (patrick-medusajs) [reply packages/medusa/src/api/routes/store/products/index.ts:134]: Make sense. I re-introduced the support for unknown fields. The typing is different but it achieves the same result and I can more easily replicate it in OAS. See 2b8779e9abf6ee11ff6decf79c48b53d8d429390

**Pull request merged**